retrofit/src/main/java/retrofit2/KotlinExtensions.kt

```diff
@@ -14,9 +14,51 @@
  * limitations under the License.
  */
 
-// Hide the class from Java consumers.
-@file:JvmName("-KotlinExtensions")
+@file:JvmName("KotlinExtensions")
 
 package retrofit2
 
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+
 inline fun <reified T> Retrofit.create(): T = create(T::class.java)
+
+suspend fun <T : Any> Call<T>.await(): T {
+  return suspendCancellableCoroutine { continuation ->
+    continuation.invokeOnCancellation {
+      cancel()
+    }
+    enqueue(object : Callback<T> {
+      override fun onResponse(call: Call<T>, response: Response<T>) {
+        if (response.isSuccessful) {
+          // TODO handle nullability
+          continuation.resume(response.body()!!)
+        } else {
+          continuation.resumeWithException(HttpException(response))
+        }
+      }
+
+      override fun onFailure(call: Call<T>, t: Throwable) {
+        continuation.resumeWithException(t)
```

Comment:
Something you may need to be aware of:

> Invocation of resumeWithException in cancelled state triggers exception handling of passed exception.

https://github.com/Kotlin/kotlinx.coroutines/issues/830

Reply:
I've been following it, yes. It's another blocker.

Reply:
No action to take because it's now ignored.